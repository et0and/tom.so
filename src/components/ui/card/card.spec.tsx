import React from "react";
import { render } from "@testing-library/react";
import {
  Card,
  CardHeader,
  CardFooter,
  CardTitle,
  CardDescription,
  CardContent,
} from "./card";

describe("<Card/>", () => {
  it("renders Card correctly", () => {
    const { asFragment } = render(<Card>Test Card</Card>);
    expect(asFragment()).toMatchSnapshot();
  });

  it("renders CardHeader correctly", () => {
    const { asFragment } = render(<CardHeader>Test Header</CardHeader>);
    expect(asFragment()).toMatchSnapshot();
  });

  it("renders CardFooter correctly", () => {
    const { asFragment } = render(<CardFooter>Test Footer</CardFooter>);
    expect(asFragment()).toMatchSnapshot();
  });

  it("renders CardTitle correctly", () => {
    const { asFragment } = render(<CardTitle>Test Title</CardTitle>);
    expect(asFragment()).toMatchSnapshot();
  });

  it("renders CardDescription correctly", () => {
    const { asFragment } = render(
      <CardDescription>Test Description</CardDescription>,
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it("renders CardContent correctly", () => {
    const { asFragment } = render(<CardContent>Test Content</CardContent>);
    expect(asFragment()).toMatchSnapshot();
  });

  it("renders full Card structure correctly", () => {
    const { asFragment } = render(
      <Card>
        <CardHeader>
          <CardTitle>Test Title</CardTitle>
          <CardDescription>Test Description</CardDescription>
        </CardHeader>
        <CardContent>Test Content</CardContent>
        <CardFooter>Test Footer</CardFooter>
      </Card>,
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
