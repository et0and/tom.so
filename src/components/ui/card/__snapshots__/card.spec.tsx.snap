// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Card/> renders Card correctly 1`] = `
<DocumentFragment>
  <div
    class="text-black"
  >
    Test Card
  </div>
</DocumentFragment>
`;

exports[`<Card/> renders CardContent correctly 1`] = `
<DocumentFragment>
  <div
    class="p-6 pt-0"
  >
    Test Content
  </div>
</DocumentFragment>
`;

exports[`<Card/> renders CardDescription correctly 1`] = `
<DocumentFragment>
  <p
    class="text-sm text-gray-500 dark:text-gray-400"
  >
    Test Description
  </p>
</DocumentFragment>
`;

exports[`<Card/> renders CardFooter correctly 1`] = `
<DocumentFragment>
  <div
    class="flex items-center p-6 pt-0"
  >
    Test Footer
  </div>
</DocumentFragment>
`;

exports[`<Card/> renders CardHeader correctly 1`] = `
<DocumentFragment>
  <div
    class="flex flex-col space-y-1.5 p-6"
  >
    Test Header
  </div>
</DocumentFragment>
`;

exports[`<Card/> renders CardTitle correctly 1`] = `
<DocumentFragment>
  <h3
    class="text-2xl font-semibold leading-none tracking-tight"
  >
    Test Title
  </h3>
</DocumentFragment>
`;

exports[`<Card/> renders full Card structure correctly 1`] = `
<DocumentFragment>
  <div
    class="text-black"
  >
    <div
      class="flex flex-col space-y-1.5 p-6"
    >
      <h3
        class="text-2xl font-semibold leading-none tracking-tight"
      >
        Test Title
      </h3>
      <p
        class="text-sm text-gray-500 dark:text-gray-400"
      >
        Test Description
      </p>
    </div>
    <div
      class="p-6 pt-0"
    >
      Test Content
    </div>
    <div
      class="flex items-center p-6 pt-0"
    >
      Test Footer
    </div>
  </div>
</DocumentFragment>
`;
